src/routes/consent.ts

```diff
@@ -1,60 +1,129 @@
 // Copyright © 2023 Ory Corp
 // SPDX-License-Identifier: Apache-2.0
-import { defaultConfig, RouteCreator, RouteRegistrator } from "../pkg"
+import { defaultConfig, logger, RouteCreator, RouteRegistrator } from "../pkg"
 import { register404Route } from "./404"
 import { oidcConformityMaybeFakeSession } from "./stub/oidc-cert"
-import {
-  AcceptOAuth2ConsentRequestSession,
-  IdentityApi,
-  OAuth2ConsentRequest,
-} from "@ory/client"
+import { AcceptOAuth2ConsentRequestSession } from "@ory/client"
 import { UserConsentCard } from "@ory/elements-markup"
 import bodyParser from "body-parser"
-import csrf from "csurf"
+import { doubleCsrf, DoubleCsrfCookieOptions } from "csrf-csrf"
+import { Request, Response, NextFunction } from "express"
 
-async function createOAuth2ConsentRequestSession(
-  grantScopes: string[],
-  consentRequest: OAuth2ConsentRequest,
-  identityApi: IdentityApi,
-): Promise<AcceptOAuth2ConsentRequestSession> {
-  // The session allows us to set session data for id and access tokens
+const cookieOptions: DoubleCsrfCookieOptions = {
+  sameSite: "lax",
+  signed: true,
+  // set secure cookies by default (recommended in production)
+  // can be disabled through DANGEROUSLY_DISABLE_SECURE_COOKIES=true env var
+  secure: true,
+  ...(process.env.DANGEROUSLY_DISABLE_SECURE_CSRF_COOKIES && {
+    secure: false,
+  }),
+}
 
-  const id_token: { [key: string]: any } = {}
+const cookieName = process.env.CSRF_COOKIE_NAME || "__Host-ax-x-csrf-token"
+const cookieSecret = process.env.CSRF_COOKIE_SECRET
 
-  if (consentRequest.subject && grantScopes.length > 0) {
-    const identity = (
-      await identityApi.getIdentity({ id: consentRequest.subject })
-    ).data
+// Sets up csrf protection
+const {
+  invalidCsrfTokenError,
+  doubleCsrfProtection, // This is the default CSRF protection middleware.
+} = doubleCsrf({
+  getSecret: () => cookieSecret || "", // A function that optionally takes the request and returns a secret
+  cookieName: cookieName, // The name of the cookie to be used, recommend using Host prefix.
+  cookieOptions,
+  ignoredMethods: ["GET", "HEAD", "OPTIONS", "PUT", "DELETE"], // A list of request methods that will not be protected.
```

Comment:
Why not `PUT` as well?

Reply:
consent only has 2 endpoints. 

`POST` and `GET`